packages/client/src/declarations/index.ts

```diff
@@ -1751,9 +1751,21 @@ export const declarationsReducer: LoopReducer<IDeclarationsState, Action> = (
       if (action.payload) {
         const userData = JSON.parse(action.payload) as IUserData
 
+        /*
+         * This is here to ensure that even if the user manages to create a new declaration
+         * before the declarations are fetched from the server, the new declaration will still be
+         * persisted. This mostly happens in E2E tests
+         */
+        const userDeclarations = userData.declarations
+        const declarationsStoredBeforeFetch = state.declarations.filter(
+          (stateDeclaration) =>
+            !userDeclarations.find(({ id }) => stateDeclaration.id === id)
+        )
```

Comment:
I noticed we also have some synchronization issues if a declaration gets updated while we have an indexedDB write in progress, after the write finishes the previous modification gets overwritten. Not related to this change, but it's similar.

Reply:
Yea :/ we need to look into those